test/sqllogictest/ldbc_bi.slt

```diff
@@ -0,0 +1,4022 @@
+      JOIN City
+        ON City.id = tf.LocationCityId
+      JOIN Country
+        ON Country.id = City.PartOfCountryId
+       AND Country.name = 'Italy'
+  )
+  , messages_of_tagclass_by_friends AS (
+    SELECT DISTINCT f.friendId
+         , Message.MessageId AS messageid
+      FROM friend_list f
+      JOIN Message
+        ON Message.CreatorPersonId = f.friendId
+      JOIN Message_hasTag_Tag
+        ON Message_hasTag_Tag.MessageId = Message.MessageId
+      JOIN Tag
+        ON Tag.id = Message_hasTag_Tag.TagId
+      JOIN TagClass
+        ON TagClass.id = Tag.TypeTagClassId
+      WHERE TagClass.name = 'Thing'
+  )
+SELECT m.friendId AS "person.id"
+     , Tag.name AS "tag.name"
+     , count(*) AS messageCount
+  FROM messages_of_tagclass_by_friends m
+  JOIN Message_hasTag_Tag
+    ON Message_hasTag_Tag.MessageId = m.MessageId
+  JOIN Tag
+    ON Tag.id = Message_hasTag_Tag.TagId
+ GROUP BY m.friendId, Tag.name
+ ORDER BY messageCount DESC, Tag.name, m.friendId
+ LIMIT 100
+----
+Explained Query:
+  Finish order_by=[#2 desc nulls_first, #1 asc nulls_last, #0 asc nulls_last] limit=100 output=[#0..=#2]
+    Return
+      Reduce group_by=[#0, #1] aggregates=[count(*)]
+        Project (#0, #8)
+          Filter (#6) IS NOT NULL
+            Join on=(eq(#0, #1, #2) AND eq(#3, #4, #5) AND #6 = #7) type=differential
+              ArrangeBy keys=[[#0]]
+                Distinct group_by=[#0]
+                  Threshold
+                    Union
+                      Distinct group_by=[#0]
+                        Project (#4)
+                          Join on=(#0 = #1 AND #2 = #3) type=differential
+                            ArrangeBy keys=[[#0]]
+                              Get l2
+                            Get l1
+                            Get l1
+                      Negate
+                        Get l2
+              ArrangeBy keys=[[#0]]
+                Distinct group_by=[#0]
+                  Project (#0)
+                    Filter (#5 = "Italy") AND (#3) IS NOT NULL
+                      Join on=(#1 = #2 AND #3 = #4) type=differential
+                        ArrangeBy keys=[[#1]]
+                          Project (#1, #8)
+                            Filter (#1) IS NOT NULL
+                              Get materialize.public.person
+                        ArrangeBy keys=[[#1]]
+                          Project (#0, #3)
+                            Filter (#0) IS NOT NULL AND (#3) IS NOT NULL
+                              Get materialize.public.city
+                        ArrangeBy keys=[[#0]]
+                          Get materialize.public.country
+              ArrangeBy keys=[[#0]]
+                Distinct group_by=[#1, #0]
+                  Project (#1, #9)
+                    Get materialize.public.message
+              ArrangeBy keys=[[#0]]
+                Distinct group_by=[#0]
+                  Project (#0)
+                    Filter (#5 = "Thing") AND (#3) IS NOT NULL
```

Comment:
I think this one _would_ get pushed and turned into a lookup if you created an index on `TagClass.name`.

Reply:
Confirmed! Good catch on the indices.